Dockerfile

```diff
@@ -650,36 +645,36 @@ COPY --from=pkg-sd-boot /*.efi.stub /sd-stub-${TARGETARCH}.efi
 FROM tools AS depmod-amd64
 WORKDIR /staging
 COPY hack/modules-amd64.txt .
-COPY --from=pkg-kernel-amd64 /lib/modules lib/modules
+COPY --from=pkg-kernel-amd64 /usr/lib/modules usr/lib/modules
 RUN <<EOF
 set -euo pipefail
 
-KERNEL_VERSION=$(ls lib/modules)
+KERNEL_VERSION=$(ls usr/lib/modules)
 
-xargs -a modules-amd64.txt -I {} install -D lib/modules/${KERNEL_VERSION}/{} /build/lib/modules/${KERNEL_VERSION}/{}
+xargs -a modules-amd64.txt -I {} install -D usr/lib/modules/${KERNEL_VERSION}/{} /build/usr/lib/modules/${KERNEL_VERSION}/{}
 
-depmod -b /build ${KERNEL_VERSION}
+depmod -b /build/usr ${KERNEL_VERSION}
 EOF
 
 FROM scratch AS modules-amd64
-COPY --from=depmod-amd64 /build/lib/modules /lib/modules
+COPY --from=depmod-amd64 /build/usr/lib/modules /usr/lib/modules
 
 FROM tools AS depmod-arm64
 WORKDIR /staging
 COPY hack/modules-arm64.txt .
-COPY --from=pkg-kernel-arm64 /lib/modules lib/modules
+COPY --from=pkg-kernel-arm64 /usr/lib/modules usr/lib/modules
 RUN <<EOF
 set -euo pipefail
 
-KERNEL_VERSION=$(ls lib/modules)
+KERNEL_VERSION=$(ls usr/lib/modules)
 
-xargs -a modules-arm64.txt -I {} install -D lib/modules/${KERNEL_VERSION}/{} /build/lib/modules/${KERNEL_VERSION}/{}
+xargs -a modules-arm64.txt -I {} install -D usr/lib/modules/${KERNEL_VERSION}/{} /build/usr/lib/modules/${KERNEL_VERSION}/{}
```

Comment:
This change would need a quirk, as it would affect how we merge kernel modules when installing extensions

Reply:
quirk in what part? and yes, this is going to be a breaking change as extensions are now disallowed to have `/lib` on their rootfs

Reply:
quirk in Talos/imager/Image Factory, as Image Factory handles any version of Talos